### PR TITLE
refactor(ooniprobe): migrate database to internal

### PR DESCRIPTION
**Pull request by DecFox**
## Checklist

- [x] I have read the [contribution guidelines](https://github.com/ooni/probe-cli/blob/master/CONTRIBUTING.md)
- [x] reference issue for this pull request: https://github.com/ooni/probe/issues/2352
- [ ] if you changed anything related how experiments work and you need to reflect these changes in the ooni/spec repository, please link to the related ooni/spec pull request: 

## Description

This diff migrates the existing database implementation to `internal`.

### CODE REVIEW

File: internal/database/actions.go [hunk 1]
```diff
@@ -254,7 +253,7 @@ func CreateMeasurement(sess db.Session, reportID sql.NullString, testName string
 func CreateResult(sess db.Session, homePath string, testGroupName string, networkID int64) (*Result, error) {
 	startTime := time.Now().UTC()
 
-	p, err := utils.MakeResultsDir(homePath, testGroupName, startTime)
+	p, err := makeResultsDir(homePath, testGroupName, startTime)
```
Comment (bassosimone) [inline internal/database/actions.go:256]: If `utils.MakeResultsDir` is now unused, can you remove its implementation form the codebase?

File: internal/database/actions.go [hunk 2]
```diff
@@ -276,7 +275,7 @@ func CreateResult(sess db.Session, homePath string, testGroupName string, networ
 }
 
 // CreateNetwork will create a new network in the network table
-func CreateNetwork(sess db.Session, loc enginex.LocationProvider) (*Network, error) {
+func CreateNetwork(sess db.Session, loc engine.LocationProvider) (*Network, error) {
```
Comment (bassosimone) [inline internal/database/actions.go:278]: Can you update the codebase to use `engine.LocationProvider` and delete `enginex.LocationProvider`?

File: internal/database/utils.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+// ResultTimestamp is a windows friendly timestamp
+const ResultTimestamp = "2006-01-02T150405.999999999Z0700"
```
Comment (bassosimone) [inline internal/database/utils.go:13]: Perhaps this should be lowercase so it's not exported by `database`?

**Review by bassosimone — CHANGES_REQUESTED** [review]
Thank you! I think we need to apply some changes before merging!

File: internal/database/actions.go [hunk 4]
```diff
@@ -84,7 +83,7 @@ func GetMeasurementJSON(sess db.Session, measurementID int64) (map[string]interf
 	}
 	// MeasurementFilePath might be NULL because the measurement from a
 	// 3.0.0-beta install
-	if measurement.Measurement.MeasurementFilePath.Valid == false {
+	if measurement.Measurement.MeasurementFilePath.Valid {
```
[the remark this answers is not in the document]
Comment (DecFox) [reply internal/database/actions.go]: Yes! thank you for pointing this out

**Comment by DecFox** [comment]
Thank you for the feedback! I have made the required changes.

File: internal/database/utils.go [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+	"path/filepath"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+// ResultTimestamp is a windows friendly timestamp
+const resultTimestamp = "2006-01-02T150405.999999999Z0700"
+
+// MakeResultsDir creates and returns a directory for the result
```
Comment (bassosimone) [inline internal/database/utils.go:15]:
```suggestion
// makeResultsDir creates and returns a directory for the result
```

**Review by bassosimone — APPROVED** [review]
:whale:

**Comment by bassosimone** [comment]
I have checked how the CI is failing and 🤷... I will merge because the failures seem unrelated and transient network issues. If I am wrong, this kind of failure will continue throughout the day, eventually forcing me to look deeper into it.

**Pull request merged**